### ISSUE TITLE
Cannot start a profession with a modded firearm in a holster.

### ISSUE THREAD

**Issue by ShadowDragon8685** [CleverRaven/Cataclysm-DDA#22440]
Game version: 0.C-6933
Operating system: Windows 7 x64
### Expected behavior
When attempting to make a new start profession, I attempted to make the profession start with a modded firearm in a holster by means of contents-item for mods and container-item for the holster. I expected the character to begin with a modded firearm in their holster.
### Actual behavior
Instead, the firearm and all of the mods for the firearm spawned, loosely, in the holster, ready to be drawn as if they were all weapons.
### Steps to reproduce the behavior

Attempt to do something similar to below in a starting profession:
                    {
                        "item": "usp_9mm",
                        "ammo-item": "9mm",
                        "charges": 15,
                        "contents-item": [
                            "match_trigger",
                            "pistol_grip",
                            "barrel_ported",
                            "laser_sight"
                        ],
                        "container-item": "holster"
                    },
Neither the order of contents-item and container-item mattered. Attempting to begin with an "item": "holster" which had the weapon as a "contents-item": failed utterly, because "contents-item": did not accept a nested "item": entry.

**Comment by Broken-Admin** [comment]
This is because you're trying to use the holster as a container. Try changing `contents-item` with `mods`

**Comment by ShadowDragon8685** [comment]
@Broken-Admin That did not work.

```
{
						"item": "sw_22",
						"ammo-item": "22_lr",
						"charges": 10,
						"container-item": "XL_holster",
						"mods": [
							"pistol_grip",
							"tuned_mechanism",
							"match_trigger",
							"waterproof_gunmod",
							"offset_sights",
							"barrel_big",
							"suppressor_compact",
							"holo_sight",
							"laser_sight"
						]
					},
```
This only yielded an unmodded weapon in a holster.

**Comment by Broken-Admin** [comment]
It's *something*-mods, I'll get help on the Discord because I forgot what it is.

**Comment by Night-Pryanik** [comment]
The only way I found to make what you want is to 
1. Add a clone of the weapon with mods installed:

```
{
    "type": "GUN",
    "id": "garand_with_mods",
    "copy-from": "garand",
    "default_mods": [ "shoulder_strap" ],
    "name": "M1 Garand"
},

```

2. Add this clone to the `entries` field of your profession:

```
"entries": [ 
    { "item": "garand_with_mods", "ammo-item": "3006", "charges": 8, "container-item": "back_holster" },
 ]
},

```

This way you'll get back holster with M1 Garand with shoulder strap within.

**Comment by Night-Pryanik** [comment]
Yeah,that's a workaround, not a proper solution, especially with tiles on.

**Comment by Night-Pryanik** [comment]
@ShadowDragon8685 look how @Firestorm01X2 solved the problem in #27239:

Create an `item_group` with `subtype: collection`, then add `entries` field where you can define your weapon, its ammo, and its mods. Then add `contents-group` to your profession instead of a `contents-item` or `container-item` with your newly added item group.
